conan/tools/microsoft/visual.py

```diff
@@ -1,7 +1,65 @@
 import os
+import textwrap
 
+from conan.tools.env.environment import register_environment_script
+from conans.client.tools import intel_compilervars_command
 from conans.client.tools.win import vs_installation_path
 from conans.errors import ConanException
+from conans.util.files import save
+
+CONAN_VCVARS_FILE = "conanvcvars.bat"
+
+
+class VCVars:
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+
+    def generate(self, auto_activate=True):
+        write_conanvcvars(self._conanfile, auto_activate=auto_activate)
+
+
+def write_conanvcvars(conanfile, auto_activate=True):
```

Comment:
I think if we are going to give the ``VCVars`` interface to users, lets use this as interface always, and hide the function ``write_conanvccars()``. Seems more consistent with other generators, and there is no issue in other generator just callling ``VCVars(conanfile).generate()`` and the ``auto_activate`` can be parameters in the constructor or attribute, to have the ``generate()`` interface match (no parameters anywhere).